scripts/run_mock_server.py

```diff
@@ -0,0 +1,350 @@
+    return HashableDict({key: _hashable_value(value) for key, value in dct.items()})
+
+
+def sanitize_msg(msg: dict) -> HashableDict:
+    """Sanitize command message."""
+    msg = msg.copy()
+    msg.pop("messageId", None)
+    return make_dict_hashable(msg)
+
+
+def add_command_response(
+    command_responses: DefaultDict[HashableDict, list],
+    record: dict,
+) -> None:
+    """Add a command response to command_responses map."""
+    command_msg = sanitize_msg(record["command_msg"])
+    # Response message doesn't need to be sanitized here because it will be sanitized
+    # in the MockZwaveJsServer.send_command_response method.
+    response_msg = record["response_msg"]
+    command_responses[command_msg].append(response_msg)
+
+
+def get_args() -> argparse.Namespace:
+    """Get arguments."""
+    parser = argparse.ArgumentParser(description="Dump Instance")
+    parser.add_argument(
+        "network_state_path", type=str, help="File path to network state dump JSON."
+    )
+    parser.add_argument("--host", type=str, help="Host to bind to", default="127.0.0.1")
+    parser.add_argument(
+        "--port", type=int, help="Port to run on (defaults to 3000)", default=3000
+    )
+    parser.add_argument(
+        "--log-level",
+        type=str.upper,
+        help="Log level",
+        default="INFO",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR"],
+    )
+    parser.add_argument(
+        "--events-to-replay-path",
+        type=str,
+        help=(
+            "File path to events to replay JSON. Events provided by --combined-replay-dump-path"
+            "option will be first, followed by events from this file."
+        ),
+        default=None,
+    )
+    parser.add_argument(
+        "--command-responses-path",
+        type=str,
+        help=(
+            "File path to command response JSON. Command responses provided by "
+            "--combined-replay-dump-path option will be first, followed by responses from this"
+            "file."
+        ),
+        default=None,
+    )
+    parser.add_argument(
+        "--combined-replay-dump-path",
+        type=str,
+        help=(
+            "File path to the combined event and command response dump JSON. Events and "
+            "command responses will be extracted in the order they were received."
+        ),
+        default=None,
+    )
+    return parser.parse_args()
+
+
+def main() -> None:
+    """Run main entrypoint."""
+    args = get_args()
+
+    with open(args.network_state_path, "r", encoding="utf8") as fp:
+        network_state_dump: List[dict] = json.load(fp)
+
+    events_to_replay = []
+    command_responses: DefaultDict[HashableDict, list] = defaultdict(list)
+
+    if args.combined_replay_dump_path:
+        with open(args.combined_replay_dump_path, "r", encoding="utf8") as fp:
+            records: List[dict] = json.load(fp)
+
+            for record in records:
+                if record["record_type"] == "event":
```

Comment:
Maybe it's enough to require the record_type key here? I guess we'll get a KeyError here already.

Reply:
I improved guards in both locations. With that being said, I think we should ignore any additional lack of guards for now and focus on functionality. We can always add in more guards later and because this is a developer tool, it's on us to make sure that both the method to generate these replay dumps and the server logic work well in tandem.